pkg/healthcheck/healthcheck_multicluster.go

```diff
@@ -394,8 +456,138 @@ func (hc *HealthChecker) checkDaisyChains() error {
 	return &SkipError{Reason: "not checking muticluster"}
 }
 
+func (hc *HealthChecker) checkLocalGateways() error {
+
+	offendingCluster, err := checkGateways(hc.kubeAPI)
+	if err != nil {
+		return err
+	}
+
+	if offendingCluster != "" {
+		return fmt.Errorf("Some gateways have problems:\n\t%s", offendingCluster)
+	}
+
+	return nil
+}
+
+func (hc *HealthChecker) checkRemoteGateways() error {
+
+	if len(hc.remoteClusterConfigs) == 0 {
+		return &SkipError{Reason: "no target cluster configs"}
+	}
+
+	var offendingClusters []string
+	for _, cfg := range hc.remoteClusterConfigs {
+		clientConfig, err := clientcmd.RESTConfigFromKubeConfig(cfg.APIConfig)
+		if err != nil {
+			offendingClusters = append(offendingClusters, fmt.Sprintf("* %s: unable to parse api config", cfg.ClusterName))
+			continue
+		}
+
+		remoteAPI, err := k8s.NewAPIForConfig(clientConfig, "", []string{}, requestTimeout)
+		if err != nil {
+			offendingClusters = append(offendingClusters, fmt.Sprintf("* %s: unable to instantiate api", cfg.ClusterName))
+			continue
+		}
+
+		offendingCluster, err := checkGateways(remoteAPI)
+		if err != nil {
+			offendingClusters = append(offendingClusters, err.Error())
+			continue
+		}
+
+		if offendingCluster != "" {
+			offendingClusters = append(offendingClusters, fmt.Sprintf("%s:\n\t%s", cfg.ClusterName, offendingCluster))
+		}
+	}
+	if len(offendingClusters) > 0 {
+		return fmt.Errorf("Some gateways have problems:\n\t%s", strings.Join(offendingClusters, "\n\t"))
+	}
+	return nil
+}
+
+func checkGateways(api *k8s.KubernetesAPI) (string, error) {
```

Comment:
this could use a comment explaining what it returns

Reply:
I think this method would be a lot easier to understand if it simply returned `[]error` instead of `(string, error)`.  

I don't think we need to build up separate slices of different error types (misconfigured ports, nonexistant gateways, etc) we can just append to a single error slice.